File: core/flowcatalyst-platform/src/main/java/tech/flowcatalyst/eventtype/EventTypeSource.java
package tech.flowcatalyst.eventtype;

/**
 * Source of an event type - how it was created.
 */
public enum EventTypeSource {
    /**
     * Created or synced via SDK/API.
     */
    API,

    /**
     * Created via the user interface.
     */
    UI
}

File: packages/platform/src/domain/event-type/event-type-source.ts
/**
 * EventType Source
 *
 * Indicates how an event type was created:
 * - CODE: Platform-defined event types
 * - API: Created via SDK/API sync
 * - UI: Created via admin UI
 */

export type EventTypeSource = "CODE" | "API" | "UI";

export const EventTypeSource = {
	CODE: "CODE" as const,
	API: "API" as const,
	UI: "UI" as const,
} as const;
